sql/core/src/main/scala/org/apache/spark/sql/execution/command/createDataSourceTables.scala

```diff
@@ -140,8 +140,8 @@ case class CreateDataSourceTableAsSelectCommand(
         return Seq.empty
       }
 
-      saveDataIntoTable(
-        sparkSession, table, table.storage.locationUri, query, mode, tableExists = true)
+      saveDataIntoTable(sparkSession, table, table.storage.locationUri, query, mode,
```

Comment:
`CreateTableAsSelectCommand` overwrite table path or not has no relation with the `SaveMode` of CTAS, that is if the table has already exist, we will overwrite it whatever the SaveMode is except that we should still check the path if exists for a managed(another [PR](https://github.com/apache/spark/pull/17057) to resolve this), so I think overwrite can't be replaced by SaveMode